### ISSUE TITLE
Refactor algorithms library and make a space for global algorithms

### ISSUE THREAD

**Issue by DraTeots** [eic/EICrecon#83]
**A problem**

Currently Tracking in algorithms includes both independent algorithms and Jana factory (generic but framework dependent). It was a temporary move while splitting algorithms to framework independent and framework dependent parts and it is time to clean this up. 

**Proposed solution**

- Change CMake to have a single `algorithms` library with both static and shared entities. 
- For now create `srd/global` directory where things like global tracking/PID/etc framework dependent factories will live (any better naming and directory structure suggestions?)
- Move tracking parts correctly between folders (see #81)
- Synchronize with current other ongoing developments

**Comment by faustus123** [comment]
So if I understand this correctly, the idea is to have:

src/algorithms  - generic, non-JANA algorithm classes
src/detectors  - detector specific JANA algorithm classes (mostly wrappers for things in src/algorithms)
src/global  - JANA algorithm classes for global reconstruction like tracking, PID, etc. (also with wrappers for src/algorithms)

It is possible that the src/algorithms directory could be replaced by the [eic/algorithms repository](https://github.com/eic/algorithms)  or by  something similar from key4hep in the future.

Is this a correct description?

**Comment by DraTeots** [comment]
Yes, eventually I also see it like this.  Certainly there should be a community defined API for switch which is not existent at this point. But it is a change exactly in this direction.

**Comment by DraTeots** [comment]
To remove naming collision I renamed algorithms libraries targets as: 

- algorithms_tracking_library
- algorithms_calorimetry_library
- algorithms_digi_library
